### PR TITLE
 replace validate_* with datatypes/assert_type

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -56,23 +56,17 @@
...
-  $install_path            = $nsclient::params::install_path
+  $allowed_hosts                  = $nsclient::params::allowed_hosts,
+  $service_state                  = $nsclient::params::service_state,
+  $service_enable                 = $nsclient::params::service_enable,
+  String $package_source_location = $nsclient::params::package_source_location,
+  String $package_source          = $nsclient::params::package_source,
+  String $package_name            = $nsclient::params::package_name,
+  $download_destination           = $nsclient::params::download_destination,
+  String $config_template         = $nsclient::params::config_template,
+  String $install_path            = $nsclient::params::install_path
```
Comment (juniorsysadmin) [inline manifests/init.pp]: Not absolute path?

File: manifests/install.pp [hunk 2]
```diff
@@ -9,10 +9,10 @@
 #
 class nsclient::install {
 
-  validate_string($nsclient::package_source_location)
-  validate_string($nsclient::package_source)
-  validate_string($nsclient::package_name)
-  validate_string($nsclient::install_path)
+  assert_type(String, $nsclient::package_source_location)
```
Comment (juniorsysadmin) [inline manifests/install.pp]: Given init.pp is the entry point, these would have already generated a failure earlier, no?

Comment (bastelfreak) [reply manifests/install.pp]: yes

Comment (bastelfreak) [reply manifests/install.pp]: updated it

File: manifests/init.pp (hunk 1 above)
Comment (bastelfreak) [reply manifests/init.pp]: I always try to only replace the old validate method. In this case it was validate_string and not validate_path

**Pull request merged**